### PR TITLE
"Add defer for locking so we don't deadlock, split up get events"

### PR DESCRIPTION
**Pull request by nikita-sirohi**

### CODE REVIEW

File: sources/github.go [hunk 1]
```diff
@@ -105,10 +105,12 @@ func (g *Github) alreadyDispatched(eventID string) bool {
 	return false
 }
 
-func (g *Github) fetchRepo(repo string) {
-	parts := strings.Split(repo, "/")
+func (g *GitHub) pullAndValidateEvents(parts []string) {
+	g.mu.Lock()
```
Comment (bregydoc) [inline sources/github.go:109]: Why the g.mu.Lock() and Unlock() are duplicated?

**Review by bregydoc — APPROVED** [review]
I think is a better approach, but I don't sure why the Mutex Lock and Unlock are duplicated in the same procedure

File: sources/github.go (hunk 1 above)
Comment (nikita-sirohi) [reply sources/github.go:109]: thanks, looks like a typo

**Comment by nikita-sirohi** [comment]
trying to figure out how to squash commit...

**Review by bregydoc — APPROVED** [review]
Looks ok

**Pull request merged**